protoc_plugin/test/send_protos_via_sendports_test.dart

```diff
@@ -15,7 +15,7 @@ Future<T> sendReceive<T>(T object) async {
   return (await rp.first) as T;
 }
 
-Future main() async {
+void main() async {
```

Comment:
I'm surprised that this works.. `main` doesn't have a return. Why is this not a type error? I thought the `async` suffix in the header means a `Future` result, and return type specifies what the returned future resolves to. So for example this function would previously return `Future<Void>`, and now it's `Future<Future>`.

Reply:
I think this is special cased in the type system - this is effectively a `Future<void>`. I'm not fully sure of that though; I did see a good explanation of this on an issue by @eernstg a while ago.

Reply:
There is actually a difference. `void f(...) async {...}` is a "fire-and-forget" asynchronous function, which is useful in the case where awaiting the completion of the execution of the body of `f` at a call site is unlikely to be useful. For instance:

```dart
void f() async {
  await null;
  print('... do some background work that nobody else should await ...');
}

void main() async {
  print('main');
  f(); // `await f()` is a compile-time error.
  await g(); // `print(await g())` is a compile-time error.
  print('end of main');
}
```

In contrast, `Future<void> g(...) async {...}` is used when it _is_ useful for a caller to synchronize with the completion of the work done in the callee, but the returned value should be ignored. So `await g(...);` is fine, but `print(await g())` is a compile-time error.

Reply:
Thanks @eernstg, that's very helpful.

I just realized that I read this as change from `void` to `Future`, but it actually changes `Future` to `void`. So I'm surprised that the current version works, not the version with this change.